### PR TITLE
fix audit #4, deprecate auto add validator

### PR DESCRIPTION
**Pull request by luciotato**
fix #4 https://github.com/neodyme-labs/2023-09-marinade-audit/issues/4

### CODE REVIEW

File: programs/marinade-finance/src/instructions/user/deposit_stake_account.rs [hunk 1]
```diff
@@ -137,87 +130,29 @@ impl<'info> DepositStakeAccount<'info> {
                 .map_err(|e| e.with_account_name("stake_account"));
         }
 
-        let validator_active_balance =
-            if validator_index == self.state.validator_system.validator_count() {
-                if self.state.validator_system.auto_add_validator_enabled == 0 {
-                    return err!(MarinadeError::AutoAddValidatorIsNotEnabled);
```
Comment (aankor) [inline programs/marinade-finance/src/instructions/user/deposit_stake_account.rs:143]: Can you deprecate it in error.rs also?

File: programs/marinade-finance/src/instructions/user/deposit_stake_account.rs [hunk 2]
```diff
@@ -137,87 +130,29 @@ impl<'info> DepositStakeAccount<'info> {
...
-                    },
-                )?;
-                0
-            } else {
-                let mut validator = self.state.validator_system.get_checked(
-                    &self.validator_list.to_account_info().data.as_ref().borrow(),
-                    validator_index,
-                    &delegation.voter_pubkey,
-                )?;
+        let validator_active_balance = {
```
Comment (aankor) [inline programs/marinade-finance/src/instructions/user/deposit_stake_account.rs:133]: Maybe replace `let validator_active_balance = { ... current_active_balance }` by `let validator_active_balance = validator.active_balance; `

**Review by aankor — CHANGES_REQUESTED** [review]

File: programs/marinade-finance/src/instructions/user/deposit_stake_account.rs (hunk 2 above)
Comment (luciotato) [reply programs/marinade-finance/src/instructions/user/deposit_stake_account.rs:133]: that means bring `let mut validator` into the general scope (now it is in a contained scope, only where it is needed)
are you ok with that?

File: programs/marinade-finance/src/instructions/user/deposit_stake_account.rs (hunk 1 above)
Comment (luciotato) [reply programs/marinade-finance/src/instructions/user/deposit_stake_account.rs:143]: ok

File: programs/marinade-finance/src/instructions/user/deposit_stake_account.rs (hunk 2 above)
Comment (luciotato) [reply programs/marinade-finance/src/instructions/user/deposit_stake_account.rs:133]: ok, changing

**Pull request closed without merging**